### PR TITLE
[DDCE-3497] Mandated nino field + add new unit tests

### PR DESCRIPTION
**Pull request by matthewwardcap**
# DDCE-3497

- 2023 Updated headers
- Test Script added
- Dependencies Updated
- Some missing tests were added back in
- Trailing Commas removed
- Mandated nino fields
- Updated nino rule
- Updated ODS test files, XML and CSV
- Updated unit tests

## PLEASE READ BEFORE REVIEW:
This change has to be left un-merged as part of the CR, this will be merged later sometime in April.

ODS files will need to be checked manually against the original as they can't be viewed in github.

Acceptance tests will need to be compared against the DDCE-3497 branch of `ers-returns-acceptance-tests` as it also can't be merged until April (feature switch)

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/services/ersTemplatesInfo/EMITemplateInfo.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ trait EMITemplateInfo {
     "3.Employee first name"	,
     "4.Employee second name(if applicable)",
     "5.Employee last name"	,
-    "6.National Insurance number(if applicable)"	,
+    "6.National Insurance number"	,
```
Comment (slavvco) [inline app/services/ersTemplatesInfo/EMITemplateInfo.scala:63]: in this file there are tabs/spaces after some " 
for example in this line
could be cleaned if there is not too much effort

File: conf/validation-config/OTHER/ers-other-notional-validation.conf [hunk 2]
```diff
@@ -106,8 +106,8 @@ ers-other-notional-validation-config{
 
     {
       column = "G"
-      cellName = "7.National Insurance number (if applicable)"
-      mandatory = false
+      cellName = "7.National Insurance number"
```
Comment (slavvco) [inline conf/validation-config/OTHER/ers-other-notional-validation.conf:109]: please add space for consistency after 7.

Comment (matthewwardcap) [reply conf/validation-config/OTHER/ers-other-notional-validation.conf:109]: Similar to ers-checking-frontend seems like the whole file is set up this way, though am happy to change it

Comment (slavvco) [reply conf/validation-config/OTHER/ers-other-notional-validation.conf:109]: please ignore as well, thanks 👍

File: app/services/ersTemplatesInfo/EMITemplateInfo.scala (hunk 1 above)
Comment (matthewwardcap) [reply app/services/ersTemplatesInfo/EMITemplateInfo.scala]: https://github.com/hmrc/ers-file-validator/pull/48/commits/d1ae8a63c65f3cf0dd49f40f09be471cf7dd28a7

Added changes

File: conf/validation-config/OTHER/ers-other-notional-validation.conf (hunk 2 above)
Comment (matthewwardcap) [reply conf/validation-config/OTHER/ers-other-notional-validation.conf]: https://github.com/hmrc/ers-file-validator/pull/48/commits/d1ae8a63c65f3cf0dd49f40f09be471cf7dd28a7

Added changes

**Review by slavvco — APPROVED** [review]

**Pull request closed without merging**